chia/wallet/wallet_node.py

```diff
@@ -225,6 +232,94 @@ async def get_private_key(self, fingerprint: Optional[int]) -> Optional[PrivateK
 
         return key
 
+    def set_resync_on_startup(self, fingerprint, enabled: bool = True):
+        with lock_and_load_config(self.root_path, "config.yaml") as config:
+            if enabled is True:
+                config["wallet"]["reset_sync_for_fingerprint"] = fingerprint
+                self.log.info("Enabled resync for wallet fingerprint: %s", fingerprint)
+            else:
+                self.log.debug(
+                    "Trying to disable resync: %s [%s]", fingerprint, config["wallet"].get("reset_sync_for_fingerprint")
+                )
+                if config["wallet"].get("reset_sync_for_fingerprint") == fingerprint:
+                    del config["wallet"]["reset_sync_for_fingerprint"]
+                    self.log.info("Disabled resync for wallet fingerprint: %s", fingerprint)
+            save_config(self.root_path, "config.yaml", config)
+
+    async def reset_sync_db(self, db_path, fingerprint):
+        conn: aiosqlite.Connection
+        # are not part of core wallet tables, but might appear later
+        ignore_tables = set(["lineage_proofs_", "sqlite_"])
+        required_tables = [
+            "coin_record",
+            "transaction_record",
+            "derivation_paths",
+            "users_wallets",
+            "users_nfts",
+            "action_queue",
+            "key_val_store",
+            "trade_records",
+            "pool_state_transitions",
+            "singleton_records",
+            "mirrors",
+            "launchers",
+            "interested_coins",
+            "interested_puzzle_hashes",
+            "unacknowledged_asset_tokens",
+            "coin_of_interest_to_trade_record",
+            "notifications",
+            "retry_store",
+        ]
+
+        async with manage_connection(db_path) as conn:
+            self.log.info("Resetting wallet sync data...")
+            rows = list(await conn.execute_fetchall("SELECT name FROM sqlite_master WHERE type='table'"))
+            try:
+                names = set([x[0] for x in rows])
+                names = names - set(required_tables)
+                for name in names:
+                    for ignore_name in ignore_tables:
+                        if name.startswith(ignore_name):
+                            break
+                    else:
+                        raise AssertionError(
+                            f"Mismatch in expected schema to reset, found unexpected table: {name}. "
+                            "Please check if you've run all migration scripts."
+                        )
```

Comment:
it would seem simpler and more straight-forward to just print this message to the log here and return `False`, rather than throwing this exception and catching it below in order to return `False` from there. There doesn't seem to be really any opportunity for other asserts to fail here, to be caught below.

You could even remove the whole `try`/`except` block and save a level of indentation